src/lib/Components/Artist/Artworks/index.tsx

```diff
@@ -19,33 +19,22 @@ interface RenderSectionParams {
   mapPropsToArtworksConnection: (Props) => any
 }
 
-interface Props extends ViewProperties {
-  artist: {
-    counts: {
-      for_sale_artworks: number
-      artworks: number
-    }
-    not_for_sale_artworks: any[]
-    for_sale_artworks: any[]
-  }
-  relay: any
+interface Props extends RelayProps, ViewProperties {
+  relay?: RelayProps
 }
 
 interface State {
   completedForSaleWorks: boolean
 }
 
-class Artworks extends React.Component<Props, State> {
-  constructor(props) {
-    super(props)
-    this.state = {
-      completedForSaleWorks: false,
-    }
+class Artworks extends Component<Props, State> {
+  state = {
+    completedForSaleWorks: false,
   }
 
   render() {
     const forSaleCount = this.props.artist.counts.for_sale_artworks
-    const otherCount = this.props.artist.counts.artworks - forSaleCount
+    const otherCount = (this.props.artist.counts.artworks as number) - (forSaleCount as number)
```

Comment:
@alloy - after fixing my typechecker and eliminating some of those top-of-the file Props overrides things got a little stricter and this one was interesting to try to understand... 

If you look [here](https://github.com/artsy/emission/blob/5430b7d70cda6dbc488bf742d007014976805cd7/src/lib/Containers/Artist.tsx#L163) and [here](https://github.com/artsy/emission/blob/5430b7d70cda6dbc488bf742d007014976805cd7/src/lib/Components/Artist/Artworks/index.tsx#L133) (this components child) and notice the types, they both say 
```
boolean | number | string | null
```
Since they're both cast to any of four types the compiler can't make basic inferences, right? And so casting it explicitly above set it through the whole chain?

Reply:
Yep, you’re code is correct. Because this value could be of a non number type (according to the typing at least), TS will complain that you can’t just subtract. Casting as a subtype, that is, one of the types in this union allows you to do so.

I don’t recall exactly why these typings are a union like that. Is the field a custom GraphQL scalar? If so, iirc those are indeed not very specific :(